kedro/pipeline/pipeline.py

```diff
@@ -160,36 +160,39 @@ def __init__(
                 "`nodes` argument of `Pipeline` is None. It must be an "
                 "iterable of nodes and/or pipelines instead."
             )
-        nodes = list(nodes)  # in case it's a generator
+        # in case it's a generator explicityly make it a list
+        # Now it can be a list of pipelines or Nodes
+        nodes_pipeline_list = list(nodes)
         _validate_duplicate_nodes(nodes)
```

Comment:
```suggestion
        _validate_duplicate_nodes(nodes_pipeline_list)
```
If it's a generator, it'd be empty here because it gets consumed on the line above. This is why I prefer leaving it as it was. 😂